scripts/microshift-cleanup-data.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+SCRIPT_NAME=$(basename $0)
+# The 'openshift-ovn-kubernetes' namespace must be in the end of the list
+# to allow for pod deletion when MicroShift is stopped
+PODS_NS_LIST=(openshift-service-ca openshift-ingress openshift-dns openshift-storage openshift-ovn-kubernetes)
+FULL_CLEAN=true
+
+function usage() {
+    echo "Stop all MicroShift services, also cleaning their data"
+    echo ""
+    echo "Usage: ${SCRIPT_NAME} <--all | --ovn>"
+    echo "   --all   Clean all MicroShift and OVN data"
+    echo "   --ovn   Clean OVN data only"
+    exit 1
+}
+
+# Parse command line
+case $1 in
+--all)
+    ;;
+--ovn)
+    FULL_CLEAN=false
+    ;;
+*)
+    usage
+    ;;
+esac
+
+# Exit if the current user is not 'root'
+if [ $(id -u) -ne 0 ] ; then
+    echo "The '${SCRIPT_NAME}' script must be run with the 'root' user privileges"
+    exit 0
+fi
+
+if ${FULL_CLEAN} ; then
+    echo "DATA LOSS WARNING: Do you wish to stop and clean ALL MicroShift data AND cri-o container workloads?"
+    select yn in "Yes" "No"; do
+        case "${yn}" in
+        Yes) 
+            break
+            ;;
+        *)
+            echo "Aborting cleanup" 
+            exit 0
+            ;;
+        esac
+    done
+fi
+
```

Comment:
Some of the rest of this script could benefit from being organized into functions. That would avoid having to check FULL_CLEAN so many times, and would make it clear to a reader what steps are going to happen in the different cleaning modes.